### PR TITLE
p11sak: Add support for the AES-XTS key type

### PR DESCRIPTION
**Pull request by ifranzki**
Add support for generating, listing and deleting AES-XTS keys.

### CODE REVIEW

File: testcases/misc_tests/p11sak_test.sh [hunk 1]
```diff
@@ -69,6 +72,13 @@ p11sak generate-key 3des --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-3des
 p11sak generate-key aes 128 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-128
 p11sak generate-key aes 192 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-192
 p11sak generate-key aes 256 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-256
+# aes-xts []128 | 256]
+if [[ -n $( pkcsconf -m -c $SLOT | grep CKM_AES_XTS_KEY_GEN) ]]; then
+	p11sak generate-key aes-xts 128 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-xts-128
+	p11sak generate-key aes-xts 256 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-xts-256
+else
+	echo "Skip generating des keys, slot does not support CKM_AES_XTS_KEY_GEN"
```
Comment (jschmidb) [inline testcases/misc_tests/p11sak_test.sh:80]: Skip generating aes-xts keys ...

File: testcases/misc_tests/p11sak_test.sh [hunk 2]
```diff
@@ -353,6 +369,81 @@ else
...
+	
+	# CK_BBOOL
+	if [[ $(grep -A 69 'p11sak-aes-xts-128' $P11SAK_AES_XTS_LONG | grep -c 'CKA_IBM_PROTKEY_EXTRACTABLE: CK_FALSE') == "2" ]]; then
+		echo "* TESTCASE list-key aes-xts PASS Listed random aes-xts public keys CK_BBOOL attribute"
+	else
+		echo "* TESTCASE list-key aes-xts FAIL Failed to list aes-xts public keys CK_BBOOL attribute"
+		status=1
+	fi
+	# CK_ULONG
+	if [[ $(grep -A 69 'p11sak-aes-xts-128' $P11SAK_AES_XTS_LONG | grep -c 'CKA_MODULUS_BITS:') == "0" ]]; then
```
Comment (jschmidb) [inline testcases/misc_tests/p11sak_test.sh:419]: What has aes-xts to do with CKA_MODULUS_BITS? Or CKA_MODULUS below? Or 'aes public key' above?
We see that this is coded in the same way also for aes, but just wondering ...

File: testcases/misc_tests/p11sak_test.sh [hunk 3]
```diff
@@ -353,6 +369,81 @@ else
...
+		echo "* TESTCASE list-key aes-xts PASS list aes-xts key pkcs#11 URI"
+	else
+		echo "* TESTCASE list-key aes-xts FAIL list aes-xts key pkcs#11 URI"
+		status=1
+	fi
+else
+	echo "* TESTCASE generate-key aes-xst-128 SKIP Generated random AES-XTS 128 key"
+	echo "* TESTCASE remove-key aes-xts-128 SKIP Deleted generated AES-TS 128 key"
+	echo "* TESTCASE generate-key aes-xst-256 SKIP Generated random AES-XTS 128 key"
+	echo "* TESTCASE remove-key aes-xts-256 SKIP Deleted generated AES-TS 128 key"
```
Comment (jschmidb) [inline testcases/misc_tests/p11sak_test.sh:443]: Typos: 
AES-TS -> AES-XTS
And in lines 440 + 441: use both times 256 (see end of line: 128)

**Comment by jschmidb** [comment]
Here a typo in commit msg:

> testcases: Fix p11sak testcase **skip** (-> SKIP in capital letters like below?) messages
> When skipping tests, output a SKIP message for all testcases that would have
> been run **it** ( -> if) it would have not been skipped.

File: man/man1/p11sak.1.in [hunk 4]
```diff
@@ -16,7 +16,7 @@ p11sak \- generate and list token keys in an openCryptoki token repository.
 
 .SH DESCRIPTION
 .B p11sak can be used to generate, list and delete the token keys in an openCryptoki token repository.
-The utility provides a flexible key management tool in openCryptoki to list and generate symmetric (DES; 3DES, AES) and asymetric (RSA, EC) keys.
+The utility provides a flexible key management tool in openCryptoki to list and generate symmetric (DES, 3DES, AES, AES-XTS) and asymetric (RSA, EC) keys.
 This tool is especially capable of a well defined listing of keys with their PKCS #11 attributes.
```
[the remark this answers is not in the document]
Comment (ifranzki) [reply man/man1/p11sak.1.in:20]: will add

File: testcases/misc_tests/p11sak_test.sh (hunk 1 above)
Comment (ifranzki) [reply testcases/misc_tests/p11sak_test.sh:80]: will fix

File: testcases/misc_tests/p11sak_test.sh (hunk 3 above)
Comment (ifranzki) [reply testcases/misc_tests/p11sak_test.sh:443]: will fix

**Comment by ifranzki** [comment]
> Here a typo in commit msg:

Will fix

**Review by vkanjur — APPROVED** [review]

**Review by jschmidb — APPROVED** [review]
Looks good!

**Pull request merged**